src/plugins/data/common/datasets/types.ts

```diff
@@ -202,6 +206,8 @@ export interface BaseDataset {
   type: string;
   /** Optional reference to the data source */
   dataSource?: DataSource;
+  /** Optional reference to the data connection */
+  dataConnection?: DataConnection;
```

Comment:
Why do we need this? cant we use the existing `dataSource` property? Event they type above indicates that they are the same. having 2 properties where only one should be populated at any time is not ideal. Lets use the `dataSource` property until we need a distinction. And if we do, we should still only have one property that can be either type so that we cant accidentally set both somehow or check for only one of the two properties. e.g. while crating the title. right now all of Discover only checks for the dataSource property and it will be really easy to miss the data connections property

Reply:
+1 we shouldn't do this.

i would check out https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/query_enhancements/public/datasets/s3_type.ts#L110 to see how we abstracted it

Reply:
good point. i kept them separated because `data-connection` is a different saved object type, there might be distinctions in the future to make them not interchangeable, and calling both types `dataSource` might be confusing. But I agree since they are mutually exclusive and are very similar now, it's better keep both under `dataSource`